### PR TITLE
HTML: support knobs for both cached and uncached nodes

### LINKED ISSUE #5017: Stories that return a DOM element don't update when knobs change
**Describe the bug**
When you adjust knobs, stories don't update when they return a DOM element but they do when you return a HTML string.  The story code is executed but is discarded due to a `forceRender` variable that is always `true`, possibly introduced in [this commit](https://github.com/storybooks/storybook/commit/5a33ea024d9e09f82ba53c8a3601a695d6e4680e#diff-a52740ffdc7b42a685aa7f8cb085d9af).

**To Reproduce**
Use the following JS for a story:

```javascript
import { storiesOf } from '@storybook/html';
import { withKnobs, text } from '@storybook/addon-knobs';

storiesOf('TestComponent', module)
  .addDecorator(withKnobs)
  .add('html', () =>  {
    let message = text('Heading', 'Hello World');
    return `<h1>${ message }</h1>`;
  })
  .add('dom', () => {
    let h1 = document.createElement('h1');
    h1.textContent = text('Heading', 'Hello World');

    return h1;
  });
```

**Expected behavior**
As you change the "Heading" knob, the stories should re-render.  In reality, only the one that returns the HTML string does.

**System:**
 - OS: macOS
 - Browser: all
 - Framework: none
 - Addons: knobs
 - Version: 4.1.1

### PR DESCRIPTION
**Pull request by Hypnosphi**
Issue: #6190 undid #4822

## What I did
Used reference comparison to fix #5017 and #6623 while keeping the effect of #4822

## How to test
– Open Knobs/DOM story, check that the knob still works
– Open Knobs/CSS transitions story, observe transitions when changing color

### CODE REVIEW

**Review by shilman — APPROVED** [review]
LGTM. 👍  Will wait for another review to merge!

**Review by ndelangen — APPROVED** [review]

**Pull request merged**
